website/blog/2022-01-06-open-sourcing-sorbet-vscode.md

```diff
@@ -0,0 +1,118 @@
+---
+id: open-sourcing-sorbet-vscode
+title: Open-sourcing the Sorbet VS Code Extension
+---
+
+We’re excited to announce that Stripe’s VS Code extension for Sorbet is now
+[open source](http://#TODO). We’ve designed Sorbet to be used in editors from
+Day 1—Sorbet exposes a flag (`--lsp`) that starts Sorbet in
+[Language Server Protocol](https://microsoft.github.io/language-server-protocol/)
+(LSP) mode. In this mode, Sorbet can respond to many LSP requests, like Go To
+Definition, Find All References, Autocomplete, and more.
+
+With this release, we’re making it even easier to take advantage of these LSP
+features when working with Sorbet in VS Code. We hope that this extension takes
+some of the guess work out of how to configure and use Sorbet’s LSP mode.
+
+<!--truncate-->
+
+Today’s release includes:
+
+- The [pre-built extension](http://#TODO) on the Visual Studio Marketplace
+- The [source code](https://github.com/sorbet/sorbet/issues/new/choose) for the
+  extension, located in the `vscode_extension/` folder of the Sorbet repo
+- Full [installation and usage instructions](https://sorbet.org/docs/vscode) in
+  the Sorbet docs
+
+## Showcase
+
+The best way to experience the Sorbet extension for VS Code is to try it out
+yourself. But to showcase a handful of its features, here are some screen
+recordings.
+
+When there are errors in a file, Sorbet will underline them in red. It also
+shows a list of all errors in VS Code’s “Problems” window.
+
+![Sorbet errors are shown with red squiggles in the editor. Hover to see the full message.](/img/lsp/errorsquiggle_blog.png)
+
+Sorbet responds with autocompletion results while typing, including a list of
+possible methods and the documentation for that item, including its signature.
+
+![Autocompletion works for constants, methods, and variables](/img/lsp/autocomplete_blog.png)
+
+Sorbet can find all references of a constant, method, or variable throughout an
+entire codebase:
+
+<video autoplay muted loop width="100%" style="display:block;margin-left:auto;margin-right:auto;">
+    <source src="/img/lsp/references_blog.mp4" type="video/mp4">
+
+    Sorry, your browser doesn't support embedded videos.
+
+</video>
+
+… and having found all those references, it can even rename them automatically:
+
+<video autoplay muted loop width="100%" style="display:block;margin-left:auto;margin-right:auto;">
+    <source src="/img/lsp/rename_blog.mp4" type="video/mp4">
+
+    Sorry, your browser doesn't support embedded videos.
+
+</video>
```

Comment:
```suggestion
<p><video autoplay muted loop width="100%" style="display:block;margin-left:auto;margin-right:auto;">
    <source src="/img/lsp/references_blog.mp4" type="video/mp4">

    Sorry, your browser doesn't support embedded videos.

</video></p>

… and having found all those references, it can even rename them automatically:

<p><video autoplay muted loop width="100%" style="display:block;margin-left:auto;margin-right:auto;">
    <source src="/img/lsp/rename_blog.mp4" type="video/mp4">

    Sorry, your browser doesn't support embedded videos.

</video></p>
```

tested this live. it makes the spacing look nicer